adv-rumors.lua

```diff
@@ -2,37 +2,30 @@
 
 --@ module=true
 
-local help = [====[
-
-adv-rumors
-==========
-Improves the "Bring up specific incident or rumor" menu in Adventure mode.
-
-- Adds all words to keywords for easier filtering/searching everywhere
-- slay/slain/slew keywords for all relevant instances
-
-]====]
-
-local adventure = df.global.game.main_interface.adventure
-
-local utils = require("utils")
-
+-- requirements
 local overlay = require('plugins.overlay')
+local utils = require('utils')
 local widgets = require('gui.widgets')
 
--- experimental, set this to 'true' to make the choices less verbose
-local shortening = false
+-- globals
+ignore_words = utils.invert{
+    "a", "an", "attacked", "by", "in", "occurred", "of", "or",
+    "s", "slain", "slew", "the", "this", "to", "was", "which"
+}
+
+-- locals
+local adventure = df.global.game.main_interface.adventure
 
 AdvRumorsOverlay = defclass(AdvRumorsOverlay, overlay.OverlayWidget)
 AdvRumorsOverlay.ATTRS{
-    desc='Widget responsible for updating the conversation screen to add keywords etc.',
-    default_pos={x=25,y=25},
+    name='adv-rumors',
```

Comment:
```suggestion
```

Reply:
`name` gets filled in by the framework and should not be set by the client (see https://docs.dfhack.org/en/latest/docs/dev/overlay-dev-guide.html#widget-attributes )

Reply:
Oooh OK!